### PR TITLE
[MPLUGIN-389] Update level to Java 8

### PR DESCRIPTION
**Pull request by slawekjaranowski**
step by step 😄

### CODE REVIEW

File: maven-plugin-plugin/src/it/ant-reference-parameter/src/main/java/org/apache/maven/plugins/antWithRefs/ArtifactsTask.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
     public void execute()
         throws BuildException
     {
-        log( String.valueOf( getProject().getReference( "artifacts" ) ) );
+        log( String.valueOf( ( Object ) getProject().getReference( "artifacts" ) ) );
```
Comment (michael-o) [inline maven-plugin-plugin/src/it/ant-reference-parameter/src/main/java/org/apache/maven/plugins/antWithRefs/ArtifactsTask.java:32]: The reference is never null?

Comment (slawekjaranowski) [reply maven-plugin-plugin/src/it/ant-reference-parameter/src/main/java/org/apache/maven/plugins/antWithRefs/ArtifactsTask.java:32]: Without such cast we will have exception:

```
Caused by: java.lang.ClassCastException: java.util.LinkedHashSet cannot be cast to [C
    at org.apache.maven.plugins.antWithRefs.ArtifactsTask.execute (ArtifactsTask.java:32)
    at org.apache.tools.ant.UnknownElement.execute (UnknownElement.java:299)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
```

As scripting Mojo are deprecated I didn't investigate it, simply did in such way to make test pass.

Comment (michael-o) [reply maven-plugin-plugin/src/it/ant-reference-parameter/src/main/java/org/apache/maven/plugins/antWithRefs/ArtifactsTask.java:32]: Yes, true. That is fully acceptable!

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #698

**Pull request merged**